stream/src/public_key/mod.rs

```diff
@@ -1,67 +1,98 @@
-//! encryption (explained below). This simple handshake is done in lieu of using
-//! TLS, Noise, WireGuard, etc. because it supports the usage of arbitrary
-//! cryptographic schemes, there is no protocol negotiation (only one way to
-//! connect), because it only takes a few hundred lines of code to implement
-//! (not having any features is a feature in safety-critical code), and because
-//! it can be simulated deterministically.
-//!
-//! In any handshake, the dialer is the party that attempts to connect to some known address/identity (public key)
-//! and the recipient of this connection is the listener. Upon forming a TCP connection, the dialer sends a signed
-//! handshake message to the listener. Besides the signature and the public key of the dialer, the handshake message
-//! contains:
-//!
-//! - The receiver's public key.
-//! - The sender's ephemeral public key (used to establish a shared secret).
-//! - The current timestamp (used to prevent replay attacks).
-//!
-//! The listener verifies the public keys are well-formatted, the timestamp is valid (not too old/not too far in the future),
-//! and that the signature is valid. If all these checks pass, the listener checks to see if it is already connected or dialing
-//! this peer. If it is, it drops the connection. If it isn't, it sends back its own signed handshake message (same as above)
-//! and considers the connection established.
-//!
-//! Upon receiving the listener's handshake message, the dialer verifies the same data as the listener and additionally verifies
-//! that the public key returned matches what they expected at the address. If all these checks pass, the dialer considers the
-//! connection established. If not, the dialer drops the connection.
-//!
-//! To better protect against malicious peers that create and/or accept connections but do not participate in handshakes,
-//! a configurable deadline is enforced for any handshake to be completed. This allows for the underlying runtime to maintain
-//! a standard read/write timeout for connections without making it easier for malicious peers to keep useless connections open.
+//! A 3-message handshake is performed to mutually authenticate and establish a shared secret.
+//! This handshake is used instead of TLS, Noise, WireGuard, etc. because:
+//! - It supports arbitrary cryptographic schemes
+//! - No protocol negotiation (only one way to connect)
+//! - Simple implementation (few hundred lines of code)
+//! - Can be simulated deterministically
+//!
+//! The **dialer** initiates connection to a known address/identity (public key)
+//! and the **listener** receives this connection. The protocol works as follows:
+//!
+//! ### Message 1: Dialer → Listener (Initial Handshake)
+//! Upon establishing a TCP connection, the dialer sends a signed handshake message to the listener.
+//! This message contains the dialer's signature and static public key, plus:
```

Comment:
nit: seems odd to include two message contents here and then a list of three others below?